### PR TITLE
Templates: Use plugincheck2 in release workflow

### PR DESCRIPTION
**Pull request by jackw**
Moving the release workflow to use plugincheck2 and the publishing config.

Fixes: #43

### CODE REVIEW

File: templates/github/ci/.github/workflows/release.yml [hunk 1]
```diff
@@ -120,10 +120,10 @@ jobs:
       - name: Lint plugin
         run: |
           git clone https://github.com/grafana/plugin-validator
-          pushd ./plugin-validator/cmd/plugincheck
+          pushd ./plugin-validator/cmd/plugincheck2
```
Comment (jackw) [inline templates/github/ci/.github/workflows/release.yml:123]: I'm confused by this path. Shouldn't it be `./plugin-validator/pkg/cmd/plugincheck2` as per the repo https://github.com/grafana/plugin-validator/tree/master/pkg/cmd/plugincheck2 ??

Comment (academo) [reply templates/github/ci/.github/workflows/release.yml:123]: yes it should be that. I wonder if this `pushd` is failing and the script goes on? Though `pushd` exits with code 1 when the path is not found.

Comment (tolzhabayev) [reply templates/github/ci/.github/workflows/release.yml:123]: It should be the path as outlined by @jackw

while we at it git clone https vs ssh?

Comment (jackw) [reply templates/github/ci/.github/workflows/release.yml]: Ok, I've updated the path. What's the benefit of using ssh over https here @tolzhabayev ? 🤔

**Review by tolzhabayev — APPROVED** [review]
LGTM!

File: templates/github/ci/.github/workflows/release.yml (hunk 1 above)
Comment (tolzhabayev) [reply templates/github/ci/.github/workflows/release.yml]: No idea, just used to `git clone` using `git@` <- ignore my comment it does not seem to add any value

**Pull request merged**